### PR TITLE
[canton-15209] Remove agreement text from 3.x

### PR DESCRIPTION
**Pull request by hubert-da**
Removal of non-ledger API-related agreement text occurrences (in 3.1).

### CODE REVIEW

**Comment by stefanobaghino-da** [comment]
IIUC the docs are frozen for 3.x, but someone from @digital-asset/daml-docs can confirm.

**Comment by hubert-da** [comment]
Based on the info, moving it to the `draft` dir.

**Comment by garyverhaegen-da** [comment]
`draft` is meant for features that are in a draft status, i.e. they're so alpha/preview/lab that we don't even want them in the main documentation. `draft` is not meant to be a draft of the next documentation for features we know will be in. That _should_ be what `3.0.0` is for, but the current version of the `3.0.0` docs is a bit of a mess and we're waiting for the Canton team to figure out how to work with docs in the 3.x era before we can work on the 3.x docs.

**Comment by hubert-da** [comment]
Thanks @garyverhaegen-da and @stefanobaghino-da. Leaving the PR in a draft then.

**Comment by garyverhaegen-da** [comment]
> > the current version of the `3.0.0` docs is a bit of a mess and we're waiting for the Canton team to figure out how to work with docs in the 3.x era before we can work on the 3.x docs.
> 
> The delay is not because of the Canton team but because we have some k8s docs in `3.0.0` that are actively used by customers and don't live anywhere else. We should have a new OSS home for them "soon."
> 
> Thanks for your patience, @hubert-da (and everyone!)

I think we have two separate issues here. On the one hand, as you explained, we cannot currently delete the existing, problematic, out-of-date `3.0.0` folder, because external people depend on it. It is an issue because it creates confusion, as in this PR.

On the other hand, if this folder were deleted right now, we also couldn't start on the "new" `3.0` folder, because the canton repo is not producing a docs bundle with its 3.x snapshots.

**Comment by garyverhaegen-da** [comment]
We have [started the 3.1 documentation folder "for real"](https://github.com/digital-asset/docs.daml.com/pull/675). It looks like this pull request was trying to make changes to 3.0.0, which has now been completely deleted. Please update it to target 3.1 instead.

File: docs/3.0.0/docs/concepts/ledger-model/ledger-integrity.rst [hunk 1]
```diff
@@ -381,21 +377,18 @@ additionally specifies, for each contract in the system:
 
 In the example, the contract model specifies that
 
-#. an `Iou obligor owner` contract has only the `obligor` as a signatory,
-   and no agreement text.
+#. an `Iou obligor owner` contract has only the `obligor` as a signatory.
```
Comment (cathyjung-da) [inline docs/3.0.0/docs/concepts/ledger-model/ledger-integrity.rst:380]:
```suggestion
#. An `Iou obligor owner` contract has only the `obligor` as a signatory.
```

File: docs/3.0.0/docs/concepts/ledger-model/ledger-integrity.rst [hunk 2]
```diff
@@ -381,21 +377,18 @@ additionally specifies, for each contract in the system:
...
 
 #. a `MustPay obligor owner` contract has both the `obligor`
```
Comment (cathyjung-da) [inline docs/3.0.0/docs/concepts/ledger-model/ledger-integrity.rst:382]:
```suggestion
#. A `MustPay obligor owner` contract has both the `obligor`
```

File: docs/3.0.0/docs/concepts/ledger-model/ledger-integrity.rst [hunk 3]
```diff
@@ -381,21 +377,18 @@ additionally specifies, for each contract in the system:
...
-   and the `owner` as signatories, with an agreement text requiring
-   the obligor to pay the owner a certain amount, off the ledger.
+   and the `owner` as signatories.
 
 #. a `PaintOffer houseOwner painter obligor refNo` contract has only the
```
Comment (cathyjung-da) [inline docs/3.0.0/docs/concepts/ledger-model/ledger-integrity.rst:385]:
```suggestion
#. A `PaintOffer houseOwner painter obligor refNo` contract has only the
```

File: docs/3.0.0/docs/concepts/ledger-model/ledger-integrity.rst [hunk 4]
```diff
@@ -381,21 +377,18 @@ additionally specifies, for each contract in the system:
...
-   painter as the signatory, with no agreement text.
+   painter as the signatory.
    Its associated key consists of the painter and the reference number.
    The painter is the maintainer.
 
 #. a `PaintAgree houseOwner painter refNo` contract has both the
```
Comment (cathyjung-da) [inline docs/3.0.0/docs/concepts/ledger-model/ledger-integrity.rst:390]:
```suggestion
#. A `PaintAgree houseOwner painter refNo` contract has both the
```

File: docs/3.0.0/docs/concepts/ledger-model/ledger-integrity.rst [hunk 5]
```diff
@@ -539,30 +532,22 @@ on the ledgers, and the information contained in contract models have
...
-* a real-world offer, whose consequences (both on- and off-ledger)
-  are specified by the **Exercise** actions on the contract allowed
-  by the contract model, and
+obligation for `A` to transfer her IOU in case she accepts the offer.
 
-* a real-world contract "proper", specified through the contract's
-  (optional) agreement text.
+Second, every contract on a Daml ledger can model a real-world offer, 
+whose consequences (both on- and off-ledger) are specified by the 
+**Exercise** actions on the contract allowed by the contract model, and
```
Comment (cathyjung-da) [inline docs/3.0.0/docs/concepts/ledger-model/ledger-integrity.rst:542]:
```suggestion
**Exercise** actions on the contract allowed by the contract model.
```

File: docs/3.0.0/docs/concepts/ledger-model/ledger-integrity.rst [hunk 6]
```diff
@@ -539,30 +532,22 @@ on the ledgers, and the information contained in contract models have
...
 
 Third, in Daml ledgers, as in the real world, one person's rights are
 another person's obligations. For example, `A`'s right to accept the
 `PaintOffer` is `P`'s obligation to paint her house in case she
 accepts.
-In Daml ledgers, a party's rights according to a contract model are the exercise actions the party can perform according to the authorization and conformance rules.
+In Daml ledgers, a party's rights according to a contract model are 
+the exercise actions the party can perform according to the authorization
```
Comment (cathyjung-da) [inline docs/3.0.0/docs/concepts/ledger-model/ledger-integrity.rst:549]:
```suggestion
the exercise actions the party can perform, based on the authorization 
```

File: docs/3.0.0/docs/concepts/ledger-model/ledger-structure.rst [hunk 7]
```diff
@@ -198,7 +198,7 @@ acceptance, the image below shows all its proper non-empty subtransactions on th
 To illustrate :doc:`contract keys </daml/reference/contract-keys>`, suppose that the contract key for a `PaintOffer` consists of the reference number and the painter.
 So Alice can refer to the `PaintOffer` by its key `(P, P123)`.
 To make this explicit, we use the notation `PaintOffer @P A &P123` for contracts, where `@` and `&` mark the parts that belong to a key.
-(The difference between `@` and `&` will be explained in the :ref:`integrity section <da-signatories-agreements-maintainers>`.)
+(The difference between `@` and `&` will be explained in the :ref:`integrity section <da-signatories-maintainers>`.)
```
Comment (cathyjung-da) [inline docs/3.0.0/docs/concepts/ledger-model/ledger-structure.rst:201]:
```suggestion
(The difference between `@` and `&` is explained in the :ref:`integrity section <da-signatories-maintainers>`.)
```

**Review by cathyjung-da — CHANGES_REQUESTED** [review]
I made a few minor suggestions. Otherwise LGTM. Thank you.

**Comment by yves-da** [comment]
If this PR is still relevant, can you @hubert-da update it and then merge. Otherwise, please close it. Thank you.

**Comment by hubert-da** [comment]
Yes, I will take care of it.

**Comment by hubert-da** [comment]
FYI, I had to reapply the changes manually due to unresolvable conflicts.

File: docs/3.1/docs/daml/reference/templates.rst [hunk 8]
```diff
@@ -260,68 +260,10 @@ The following types are *not serializable*, and thus may not be used in template
...
-Within a project where the warning has been turned off via the ``daml.yaml``
-file, it can be turned back on for individual Daml files by adding the following
-line at the top of each file:
-
-.. code-block:: daml
-
-  {-# OPTIONS_GHC -Wtemplate-agreement #-}
-
-.. _daml-ref-preconditions:
+Users should remove any agreement declarations from their code, as this feature has been fully removed from the language.
```
Comment (hubert-da) [inline docs/3.1/docs/daml/reference/templates.rst:266]: This is a new change.

File: docs/3.1/docs/tools/navigator/index.rst [hunk 9]
```diff
@@ -132,8 +132,7 @@ to allocate those parties before you can log in as them.
 Viewing Templates or Contracts
 ******************************
 
-Daml *contract ​templates* are ​models ​that contain ​the ​agreement ​statement, ​all ​the ​applicable
-parameters, ​and ​the ​choices ​that ​can ​be ​made ​in ​acting ​on ​that ​data.
+Daml *contract ​templates* are ​models ​that contain ​all ​the ​applicable parameters ​and ​the ​choices ​that ​can ​be ​made ​in ​acting ​on ​that ​data.
```
Comment (hubert-da) [inline docs/3.1/docs/tools/navigator/index.rst:135]: This is new as well.

File: docs/3.1/docs/concepts/ledger-model/ledger-daml.rst [hunk 10]
```diff
@@ -11,7 +11,6 @@ a contract model, and such a model must specify:
 
 #. a set of allowed actions on the contracts, and
```
Comment (cathyjung-da) [inline docs/3.1/docs/concepts/ledger-model/ledger-daml.rst:12]:
```suggestion
#. A set of allowed actions on the contracts
```

File: docs/3.1/docs/concepts/ledger-model/ledger-daml.rst [hunk 11]
```diff
@@ -11,7 +11,6 @@ a contract model, and such a model must specify:
...
 #. the signatories, contract observers, and
```
Comment (cathyjung-da) [inline docs/3.1/docs/concepts/ledger-model/ledger-daml.rst:13]:
```suggestion
#. The signatories, contract observers
```

File: docs/3.1/docs/concepts/ledger-model/ledger-daml.rst [hunk 12]
```diff
@@ -11,7 +11,6 @@ a contract model, and such a model must specify:
...
-#. an optional agreement text associated with each contract, and
 #. the optional key associated with each contract and its maintainers.
```
Comment (cathyjung-da) [inline docs/3.1/docs/concepts/ledger-model/ledger-daml.rst:14]:
```suggestion
#. The optional key associated with each contract and its maintainers
```

**Review by cathyjung-da — CHANGES_REQUESTED** [review]

**Review by cathyjung-da — APPROVED** [review]

**Pull request merged**